### PR TITLE
Hide controller for videos with no source

### LINKED ISSUE #236: Controller lifetime should be tied to src on video (no src, no controller)
There is doubling of controls on vk.com video player.
I am using  0.4.8 version of plugin.
Here is screenshot.
sy8ooncvsbcairmikjhkgw
You can reproduce this behavior by playing some videos on vk.com pages, for example https://vk.com/iskin

### PR DESCRIPTION
**Pull request by jacobcolbert**
Fixes #236 

Creates a mutation observer that will detect when source changes. Hides visible controller when there is no source present. Prevents double/more controllers being visible.

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -152,7 +154,26 @@
...
+      var observer=new MutationObserver((mutations)=> {
+        mutations.forEach((mutation) => {
+          if (mutation.type === 'attributes' && mutation.attributeName === 'src'){
+            var controller = document.querySelector(`div[data-vscid="${this.id}"]`);
+            if(!controller){
+              return;
+            }
+            if (!mutation.target.src) {
+              controller.classList.add('vsc-nosource');
+              console.log('Hiding: no source');
```
Comment (igrigorik) [inline inject.js]: Are the console.log's necessary? 

Sidenote: we should probably keep them around, as they can be helpful in debugging, but we should probably make a helper function (separate PR) that checks for a debug flag or some such, that can be set in development but does not otherwise need to fire for all users.

File: inject.js [hunk 2]
```diff
@@ -123,6 +123,8 @@
       } else {
         tc.settings.speeds[target.src] = tc.settings.lastSpeed;
       }
+      target.playbackRate = tc.settings.speeds[target.src];
```
Comment (igrigorik) [inline inject.js]: Why is this necessary in context of this PR?

Comment (jacobcolbert) [reply inject.js]: This is just reorganization of line 155 moved up to here. It makes sense to have all speed initialization things together and all Listeners grouped together.

Comment (jacobcolbert) [reply inject.js]: With the big new mutation block it seemed even more out of place

File: inject.js [hunk 3]
```diff
@@ -152,7 +154,26 @@
         }
       }.bind(this));
 
-      target.playbackRate = tc.settings.speeds[target.src];
+      var observer=new MutationObserver((mutations)=> {
```
[the remark this answers is not in the document]
Comment (jacobcolbert) [reply inject.js]: working on it! I come from eschewing all spaces, so still sometimes slip...

File: inject.js (hunk 1 above)
Comment (jacobcolbert) [reply inject.js]: Logs clearly not necessary. Forgot to take them out from  debugging and was gonna commit again to remove them but thought maybe they were worth keeping as they shouldn't fire too often. Probably not worth keeping without  a debug logger function.

File: inject.js (hunk 2 above)
Comment (igrigorik) [reply inject.js]: 👍
